File: node/profanity.js
// Calls the mock profanity-check stub with a timeout and fails open: any
// error or timeout is treated as "clean" and logged, so a down dependency
// never blocks a user's write (plan.md criterion #6). Mirrors
// go/internal/profanity/profanity.go. Uses the global fetch (stable since
// Node 18) — no HTTP client dependency needed.
'use strict';

class Checker {
  constructor(url, timeoutMs) {
    this.url = url;
    this.timeoutMs = timeoutMs;
  }

  async isClean(text) {
    let resp;
    try {
      resp = await fetch(`${this.url}/check`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ text }),
        signal: AbortSignal.timeout(this.timeoutMs),
      });
    } catch (err) {
      console.warn('profanity check unreachable, allowing content:', err.message);
      return true;
    }

    if (!resp.ok) {
      console.warn('profanity check returned non-200, allowing content:', resp.status);
      return true;
    }

    try {
      const result = await resp.json();
      return Boolean(result.clean);
    } catch (err) {
      console.warn('profanity check response unreadable, allowing content:', err.message);
      return true;
    }
  }
}

module.exports = { Checker };

File: python/profanity.py
"""Calls the mock profanity-check stub with a timeout and fails open: any
error or timeout is treated as "clean" and logged, so a down dependency
never blocks a user's write (plan.md criterion #6). Mirrors
go/internal/profanity/profanity.go.
"""
import logging

import httpx

log = logging.getLogger("profanity")


class Checker:
    def __init__(self, url: str, timeout: float):
        self.url = url
        self.timeout = timeout

    async def is_clean(self, text: str) -> bool:
        try:
            async with httpx.AsyncClient(timeout=self.timeout) as client:
                resp = await client.post(f"{self.url}/check", json={"text": text})
        except httpx.HTTPError as e:
            log.warning("profanity check unreachable, allowing content: %s", e)
            return True

        if resp.status_code != 200:
            log.warning("profanity check returned non-200, allowing content: %s", resp.status_code)
            return True

        try:
            return bool(resp.json()["clean"])
        except (ValueError, KeyError) as e:
            log.warning("profanity check response unreadable, allowing content: %s", e)
            return True
